Merge conflicts

```
##[error]Error: Error for {"ownerName":"Label305","repoName":"AutoRebase","number":1,"mergeableState":"behind"}: HttpError: Merge conflict
```

https://github.com/Label305/AutoRebase/pull/12/checks?check_run_id=737485272#step:5:29

--- a/.idea/compiler.xml
+++ b/.idea/compiler.xml
@@ -1,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="TypeScriptCompiler">
+    <option name="showAllErrors" value="true" />
+  </component>
+</project>

--- a/src/EligiblePullRequests/__tests__/testableEligiblePullRequestsRetriever.test.ts
+++ b/src/EligiblePullRequests/__tests__/testableEligiblePullRequestsRetriever.test.ts
@@ -25,13 +25,14 @@ test('Without open pull requests there are no eligible pull requests', async () 
 });
 
 describe('A pull request is eligible', () => {
-    it("when the mergeableState is 'behind' and it has the label 'opt-in:autorebase'", async () => {
+    it("when it is rebaseable, the mergeableState is 'behind' and it has the label 'opt-in:autorebase'", async () => {
         /* Given */
         testOpenPullRequestsProvider.openPullRequestsValue = [
             {
                 ownerName: 'owner',
                 repoName: 'repo',
                 number: 3,
+                rebaseable: true,
                 mergeableState: 'behind',
                 labels: ['opt-in:autorebase'],
             },
@@ -46,6 +47,7 @@ describe('A pull request is eligible', () => {
                 ownerName: 'owner',
                 repoName: 'repo',
                 number: 3,
+                rebaseable: true,
                 mergeableState: 'behind',
                 labels: ['opt-in:autorebase'],
             },
@@ -54,6 +56,26 @@ describe('A pull request is eligible', () => {
 });
 
 describe('A pull request is not eligible', () => {
+    it("when it isn't rebaseable", async () => {
+        /* Given */
+        testOpenPullRequestsProvider.openPullRequestsValue = [
+            {
+                ownerName: 'owner',
+                repoName: 'repo',
+                number: 3,
+                rebaseable: false,
+                mergeableState: 'behind',
+                labels: ['opt-in:autorebase'],
+            },
+        ];
+
+        /* When */
+        const results = await retriever.findEligiblePullRequests('owner', 'repo');
+
+        /* Then */
+        expect(results).toStrictEqual([]);
+    });
+
     each([['blocked'], ['clean'], ['dirty'], ['unknown'], ['unstable']]).it(
         "when the mergeableState is '%s'",
         async (mergeableState: MergeableState) => {
@@ -63,6 +85,7 @@ describe('A pull request is not eligible', () => {
                     ownerName: 'owner',
                     repoName: 'repo',
                     number: 3,
+                    rebaseable: true,
                     mergeableState: mergeableState,
                     labels: ['opt-in:autorebase'],
                 },
@@ -83,6 +106,7 @@ describe('A pull request is not eligible', () => {
                 ownerName: 'owner',
                 repoName: 'repo',
                 number: 3,
+                rebaseable: true,
                 mergeableState: 'behind',
                 labels: [],
             },

--- a/src/EligiblePullRequests/testableEligiblePullRequestsRetriever.ts
+++ b/src/EligiblePullRequests/testableEligiblePullRequestsRetriever.ts
@@ -20,7 +20,7 @@ export class TestableEligiblePullRequestsRetriever implements EligiblePullReques
         debug(`Found ${pullRequests.length} open pull requests.`);
 
         const results = pullRequests.filter((value) => {
-            return value.mergeableState === 'behind' && value.labels.includes('opt-in:autorebase');
+            return value.rebaseable && value.mergeableState === 'behind' && value.labels.includes('opt-in:autorebase');
         });
 
         debug(`${results.length} pull requests are eligible.`);

--- a/src/Github/Api/getPullRequestService.ts
+++ b/src/Github/Api/getPullRequestService.ts
@@ -6,7 +6,9 @@ export interface GetPullRequestService {
 }
 
 export interface ApiGetPullRequest {
+    rebaseable: boolean;
     mergeableState: MergeableState | undefined;
+    labels: string[];
 }
 
 export class GithubGetPullRequestService implements GetPullRequestService {
@@ -24,7 +26,9 @@ export class GithubGetPullRequestService implements GetPullRequestService {
         });
 
         return {
+            rebaseable: result.data.rebaseable,
             mergeableState: result.data.mergeable_state as MergeableState,
+            labels: result.data.labels.map((label) => label.name),
         };
     }
 }

--- a/src/Github/__tests__/githubPullRequestInfoProvider.test.ts
+++ b/src/Github/__tests__/githubPullRequestInfoProvider.test.ts
@@ -1,4 +1,4 @@
-import {GithubMergeableStateProvider} from '../githubMergeableStateProvider';
+import {GithubPullRequestInfoProvider} from '../githubPullRequestInfoProvider';
 import {ApiGetPullRequest, GetPullRequestService} from '../Api/getPullRequestService';
 import each from 'jest-each';
 
@@ -11,41 +11,47 @@ class TestGetPullRequestService implements GetPullRequestService {
 }
 
 const getPullRequestService = new TestGetPullRequestService();
-const provider = new GithubMergeableStateProvider(getPullRequestService);
+const provider = new GithubPullRequestInfoProvider(getPullRequestService);
 
-describe('The mergeableState is propagated', () => {
+describe('The pull request info is propagated', () => {
     each([['behind'], ['blocked'], ['clean'], ['dirty'], ['unstable']]).it(
         "when the mergeableState is '%s'",
         async (mergeableState) => {
             /* Given */
             getPullRequestService.results.push({
+                rebaseable: true,
                 mergeableState: mergeableState,
+                labels: [],
             });
 
             /* When */
-            const result = await provider.mergeableStateFor('owner', 'repo', 3);
+            const result = await provider.pullRequestInfoFor('owner', 'repo', 3);
 
             /* Then */
-            expect(result).toBe(mergeableState);
+            expect(result.mergeableState).toBe(mergeableState);
         },
     );
 });
 
-describe('The mergeableState is retried', () => {
+describe('The pull request info is retried', () => {
     each([['unknown'], ['invalid']]).it("when the mergeableState is '%s'", async (mergeableState) => {
         /* Given */
         getPullRequestService.results.push({
+            rebaseable: true,
             mergeableState: mergeableState,
+            labels: [],
         });
 
         getPullRequestService.results.push({
+            rebaseable: true,
             mergeableState: 'behind',
+            labels: [],
         });
 
         /* When */
-        const result = await provider.mergeableStateFor('owner', 'repo', 3);
+        const result = await provider.pullRequestInfoFor('owner', 'repo', 3);
 
         /* Then */
-        expect(result).toBe('behind');
+        expect(result.mergeableState).toBe('behind');
     });
 });

--- a/src/Github/githubOpenPullRequestsProvider.ts
+++ b/src/Github/githubOpenPullRequestsProvider.ts
@@ -1,13 +1,13 @@
 import {OpenPullRequestsProvider} from '../EligiblePullRequests/testableEligiblePullRequestsRetriever';
 import {PullRequestInfo} from '../pullrequestinfo';
 import {mapAsync} from '../utils';
-import {GithubMergeableStateProvider} from './githubMergeableStateProvider';
+import {GithubPullRequestInfoProvider} from './githubPullRequestInfoProvider';
 import {ApiListPullRequest, ListPullRequestsService} from './Api/listPullRequestsService';
 
 export class GithubOpenPullRequestsProvider implements OpenPullRequestsProvider {
     constructor(
         private listPullRequestsService: ListPullRequestsService,
-        private mergeableStateProvider: GithubMergeableStateProvider,
+        private mergeableStateProvider: GithubPullRequestInfoProvider,
     ) {}
 
     async openPullRequests(ownerName: string, repoName: string): Promise<PullRequestInfo[]> {
@@ -21,19 +21,20 @@ export class GithubOpenPullRequestsProvider implements OpenPullRequestsProvider 
     private async pullRequestInfoFor(
         ownerName: string,
         repoName: string,
-        pullRequest: ApiListPullRequest,
+        apiListPullRequest: ApiListPullRequest,
     ): Promise<PullRequestInfo> {
-        const mergeableState = await this.mergeableStateProvider.mergeableStateFor(
+        const pullRequest = await this.mergeableStateProvider.pullRequestInfoFor(
             ownerName,
             repoName,
-            pullRequest.number,
+            apiListPullRequest.number,
         );
 
         return {
             ownerName: ownerName,
             repoName: repoName,
             number: pullRequest.number,
-            mergeableState: mergeableState,
+            rebaseable: pullRequest.rebaseable,
+            mergeableState: pullRequest.mergeableState,
             labels: pullRequest.labels,
         };
     }

--- a/src/Github/githubPullRequestInfoProvider.ts
+++ b/src/Github/githubPullRequestInfoProvider.ts
@@ -1,16 +1,20 @@
 import {GetPullRequestService} from './Api/getPullRequestService';
-import {MergeableState, mergeableStates} from '../pullrequestinfo';
+import {mergeableStates, PullRequestInfo} from '../pullrequestinfo';
 import {debug} from '@actions/core';
 import {promiseRetry} from '../Util/promiseRetry';
 
-export class GithubMergeableStateProvider {
+export class GithubPullRequestInfoProvider {
     constructor(private getPullRequestService: GetPullRequestService) {}
 
-    public mergeableStateFor(ownerName: string, repoName: string, pullRequestNumber: number): Promise<MergeableState> {
-        return promiseRetry<MergeableState>(
-            async (): Promise<MergeableState> => {
+    public pullRequestInfoFor(
+        ownerName: string,
+        repoName: string,
+        pullRequestNumber: number,
+    ): Promise<PullRequestInfo> {
+        return promiseRetry<PullRequestInfo>(
+            async (): Promise<PullRequestInfo> => {
                 try {
-                    const {mergeableState} = await this.getPullRequestService.getPullRequest(
+                    const {rebaseable, mergeableState, labels} = await this.getPullRequestService.getPullRequest(
                         ownerName,
                         repoName,
                         pullRequestNumber,
@@ -25,8 +29,17 @@ export class GithubMergeableStateProvider {
                         throw Error("mergeableState is 'unknown'");
                     }
 
+                    debug(`rebaseable value for pull request #${pullRequestNumber}: ${String(rebaseable)}`);
                     debug(`mergeableState for pull request #${pullRequestNumber}: ${mergeableState}`);
-                    return mergeableState;
+
+                    return {
+                        ownerName: ownerName,
+                        repoName: repoName,
+                        number: pullRequestNumber,
+                        rebaseable: rebaseable,
+                        mergeableState: mergeableState,
+                        labels: labels,
+                    };
                 } catch (error) {
                     debug(
                         `Fetching mergeableState for pull request #${pullRequestNumber} failed: "${String(

--- a/src/index.ts
+++ b/src/index.ts
@@ -5,7 +5,7 @@ import {EligiblePullRequestsRetriever} from './EligiblePullRequests/eligiblePull
 import {Rebaser} from './rebaser';
 import {TestableEligiblePullRequestsRetriever} from './EligiblePullRequests/testableEligiblePullRequestsRetriever';
 import {GithubOpenPullRequestsProvider} from './Github/githubOpenPullRequestsProvider';
-import {GithubMergeableStateProvider} from './Github/githubMergeableStateProvider';
+import {GithubPullRequestInfoProvider} from './Github/githubPullRequestInfoProvider';
 import {GithubGetPullRequestService} from './Github/Api/getPullRequestService';
 import {GithubListPullRequestsService} from './Github/Api/listPullRequestsService';
 
@@ -15,7 +15,7 @@ async function run(): Promise<void> {
         const eligiblePullRequestsRetriever: EligiblePullRequestsRetriever = new TestableEligiblePullRequestsRetriever(
             new GithubOpenPullRequestsProvider(
                 new GithubListPullRequestsService(github),
-                new GithubMergeableStateProvider(new GithubGetPullRequestService(github)),
+                new GithubPullRequestInfoProvider(new GithubGetPullRequestService(github)),
             ),
         );
         const rebaser = new Rebaser(github);

--- a/src/pullrequestinfo.ts
+++ b/src/pullrequestinfo.ts
@@ -2,6 +2,7 @@ export type PullRequestInfo = {
     ownerName: string;
     repoName: string;
     number: number;
+    rebaseable: boolean;
     mergeableState: MergeableState;
     labels: string[];
 };

